src/components/AbilityScoreField/AbilityScoreField.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * @file A custom element that represents an Ability Score and its modifier
+ */
+import BaseElement from "../_BaseElement";
+// @ts-expect-error
+import styles from './AbilityScoreField.css' assert { type: "css" };
+import { component } from "../decorators";
+
+/**
+ * @ignore until we find a way to get Foundry to shift focus
+ *         into a component after a sheet updates/rerenders.
+ */
+@component('uft-ability-score-field')
+export default class AbilityScoreField extends BaseElement {
+  static get styles(): CSSStyleSheet[] {
+    return [styles];
+  }
+
+  protected events(): void {
+    this.shadowRoot
+      ?.querySelector(".score-field")
+      ?.addEventListener("change", (e) => {
+        this.onInput(e);
+      });
+
+    if (!this.hasAttribute("readonly") && !this.hasAttribute("disabled"))
+      this.shadowRoot
+        ?.querySelector("label")
+        ?.addEventListener("click", (e) => {
+          const evt = new Event("roll") as Event & {metaKey: boolean, ctrlKey: boolean};
```

Comment:
what is the reasoning here? Why do we check if the ability score is editable before allowing rolls on it? When is it disabled?